app/src/main/java/com/x8bit/bitwarden/data/auth/datasource/disk/AuthDiskSourceImpl.kt

```diff
@@ -122,11 +123,24 @@ class AuthDiskSourceImpl(
         storeMasterPasswordHash(userId = userId, passwordHash = null)
         storePolicies(userId = userId, policies = null)
         storeAccountTokens(userId = userId, accountTokens = null)
+        storeShouldUseKeyConnector(userId = userId, shouldUseKeyConnector = null)
 
         // Do not remove the DeviceKey or PendingAuthRequest on logout, these are persisted
         // indefinitely unless the TDE flow explicitly removes them.
     }
 
+    override fun getShouldUseKeyConnector(userId: String): Boolean =
+        requireNotNull(
```

Comment:
Do we do this with other persisted boolean values?

Reply:
Just in the `getShouldTrustDevice`.

Since the default is `false` it should never return null.